src/profiler/components/Table.tsx

```diff
@@ -44,16 +53,33 @@ const Table = ({ data, height }: ChartTableProps): JSX.Element => {
     return acc
   }, {})
 
-  const sortedData = Object.values(groupedDataByItem)
-  sortedData.sort((a: any, b: any) => {
-    const aCounts = a.map((item: any) => item.count)
-    const bCounts = b.map((item: any) => item.count)
-    const aMax = Math.max.apply(null, aCounts)
-    const bMax = Math.max.apply(null, bCounts)
-    return bMax - aMax
+  let maxIndex = 0
+
+  const tableData = Object.keys(groupedDataByItem).map(key => {
+    return groupedDataByItem[key].reduce((acc: any, item: any, index: number) => {
+      if (maxIndex < index) {
+        maxIndex = index
+      }
+
+      return {
+        ...acc,
+        key: item.key,
+        item: item.item,
+        [`count${index}`]: item.count,
+        [`color${index}`]: item.color,
+      }
+    }, {})
   })
 
-  const tableData: any = sortedData.flat().slice(0, 10)
+  const countColumns = new Array(maxIndex + 1).fill(1).map((value, index) => {
```

Comment:
In this case, we need to change here `maxIndex + 1` to `maxIndex`.
And would be better to rename `maxIndex` to `maxDataLength` or something similar (with the word 'length')